trl/trainer/sft_trainer.py

```diff
@@ -339,18 +339,18 @@ def make_inputs_require_grad(module, input, output):
             if formatting_func is not None:
                 args.dataset_kwargs["add_special_tokens"] = False
 
-        if not args.packing:
-            # If we aren't skipping data preparation, then a dataset_text_field
-            # or formatting_func must be provided.
-            if (
-                args.dataset_text_field is None
-                and formatting_func is None
-                and not args.dataset_kwargs.get("skip_prepare_dataset", False)
-            ):
-                raise ValueError(
-                    "You passed `packing=False` to the SFTTrainer/SFTConfig, but you didn't pass a `dataset_text_field` or `formatting_func` argument."
-                )
+        # If we aren't skipping data preparation, then a dataset_text_field or formatting_func must be provided.
+        if (
+            args.dataset_text_field is None
+            and formatting_func is None
+            and not args.dataset_kwargs.get("skip_prepare_dataset", False)
```

Comment:
Personally I find the double negatives hard to parse - would this be equivalent?
```suggestion
            and args.dataset_kwargs.get("skip_prepare_dataset", True)
```